oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java

```diff
@@ -421,15 +426,60 @@ private void setNameAndDescription(FixerResponse resp, String textToSearch, int
 	}
   }
 
-  private String removeOldIntegLink(String str, int level, String method) {
-    final String start = "<td><table";
-    final String end = "</table></td></tr>";
-     final String toRemove = StringUtils.substringBetween(str, start, end);
-     if(toRemove.contains("href=\"/webapps/dych-tle-")) {
-       return StringUtils.replace(str,start+toRemove+end, "<td/></tr>");
-	 } else {
-       log(level+1, "Not removing old integration link from the body - could not find the 'dych-tle' table");
-       return str;
-	 }
+  private String scrubDescription(String str, int level, String[] urls) {
+    Document d = Jsoup.parse(str);
+    d.outputSettings().prettyPrint(false);
+
+    // Remove all old integration links
+    Elements elems = d.select("a[href]");
+    for(Element e : elems) {
+      if (e.attr("href").startsWith("/webapps/dych-tle-")) {
+        e.remove();
+        log(level + 1, "Removing the element ([lt] == <): " + e.outerHtml().replaceAll("<", "[lt]"));
+	  }
+	}
+
+	// Remove all old image spacers
+	elems = d.select("img");
```

Comment:
lower down it looks like it wants to focus just in images inside tables
```
<tr><td><img .../></td></tr>
```

would it make sense to do that selection here?

```css
tr > td > img
```
https://developer.mozilla.org/en-US/docs/Web/CSS/Child_combinator

Reply:
that was a nicety, but if it can't determine that the `img` is only inside a `td` and `tr`, it'll still remove the `img`

Reply:
ahh, good to know